cluster-autoscaler/cloudprovider/test/test_cloud_provider.go

```diff
@@ -250,11 +251,16 @@ func (tcp *TestCloudProvider) NewNodeGroup(machineType string, labels map[string
 		machineType:     machineType,
 		labels:          labels,
 		taints:          taints,
+		scaleDownPolicy: deallocate.Delete,
 	}, nil
 }
 
 // BuildNodeGroup returns a test node group.
-func (tcp *TestCloudProvider) BuildNodeGroup(id string, min, max, size int, exists bool, autoprovisioned bool, machineType string, opts *config.NodeGroupAutoscalingOptions) *TestNodeGroup {
+func (tcp *TestCloudProvider) BuildNodeGroup(id string, min, max, size int, exists bool, autoprovisioned bool, machineType string, opts *config.NodeGroupAutoscalingOptions, deallocateMode bool) *TestNodeGroup {
```

Comment:
minor: 
```suggestion
func (tcp *TestCloudProvider) BuildNodeGroup(id string, min, max, size int, exists bool, autoprovisioned bool, machineType string, opts *config.NodeGroupAutoscalingOptions, scaleDownPolicy DeallocateMode) *TestNodeGroup {
```

Easier to read at the call site?

Reply:
Fixed, and added a wrapper with a default value to not have to change original call sites (to minimize delta with upstream)